command.c

```diff
@@ -1592,36 +1592,24 @@ static void perform_loadhigh(const char *arg)
   __dpmi_int(0x21, &r);
   }
 
-static void perform_loadfix(const char *arg)
-  {
-  const int numblocks = 16;
-  int orig_strat, orig_umblink, allocated, ii;
-  unsigned short allocations[numblocks], allocation, to_64kib, max, size;
+
+#define LOADFIX_NUMBLOCKS 16
+const int loadfix_numblocks = LOADFIX_NUMBLOCKS;
+unsigned short loadfix_allocations[LOADFIX_NUMBLOCKS];
+int loadfix_ii;
+unsigned loadfix_is_v;
+unsigned loadfix_initialised = 0;
+
+static void loadfix_init(unsigned is_v);
+static void loadfix_init(unsigned is_v) {
+  int orig_strat, orig_umblink, allocated;
+  unsigned short allocation, to_64kib, max, size;
   __dpmi_regs r = {};
-  int is_v = 0;
-  while (*arg != '\0')
-    {
-    if (*cmd_switch == '\0') // if not a command switch ...
-      {
-      break;
-      }
-    else
-      {
-      if (stricmp(cmd_switch,"/v")==0)
-        {
-        is_v = 1;
-        }
-      else
-        {
-        cprintf("Invalid switch - %s\r\n", cmd_switch);
-        reset_batfile_call_stack();
-        return;
-        }
-      }
-    advance_cmd_arg();
-    }
-  strcpy(cmd, arg);
-  advance_cmd_arg();
+
+  if (loadfix_initialised)
```

Comment:
Do you envision any recursion
here?

Reply:
If you do an explicit LOADFIX then loadfix_init is called twice. We want it not to overwrite / truncate the first run's entries in the array however.

Reply:
Ah I guess you actually do -
this can be called from both
perform_loadfix() and
perform_external_cmd()...
Maybe instead add another
arg to perform_external_cmd()
to suppress auto-loadfix?

Reply:
I think it is cleaner this way, no need to create chaos by changing `perform_external_cmd` parameters.